### PR TITLE
[data] Add mosaicml-streaming to BYOD requirements for release testing

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

Adds a dep required by #37610.

### CODE REVIEW

**Review by c21 — APPROVED** [review]

File: release/ray_release/byod/requirements_byod_3.8.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (can-anyscale) [inline release/ray_release/byod/requirements_byod_3.8.txt:2]: oh, you should compile this with python 3.8, not 3.9

File: release/ray_release/byod/requirements_byod_3.8.in [hunk 2]
```diff
@@ -12,6 +12,7 @@ gym[atari]
 importlib-metadata
 lightgbm
 memray
+mosaicml-streaming
```
Comment (can-anyscale) [inline release/ray_release/byod/requirements_byod_3.8.in:15]: this should be added to requirements_byod_ml_3.8.in since your test uses ray-ml image

**Review by can-anyscale — CHANGES_REQUESTED** [review]
The package needs to be added and compiled for requirements_ml_byod_3.8.in with python 3.8

**Comment by stephanie-wang** [comment]
> The package needs to be added and compiled for requirements_ml_byod_3.8.in with python 3.8

Ah thanks. Can you review again?

**Review by can-anyscale — APPROVED** [review]

**Pull request merged**